R/graded.R

```diff
@@ -650,15 +650,15 @@ fail_if_code_feedback <- function(
     return(grade_code_is_empty())
   }
 
-  if (is_placeholder(solution_code, ".solution_code")) {
-    solution_code <- get_from_env(".solution_code", env)
-    assert_object_found_in_env(solution_code, env, "fail_if_code_feedback", throw_grade = FALSE)
+  if (is_placeholder(solution_code_all, ".solution_code_all")) {
+    solution_code_all <- get_from_env(".solution_code_all", env)
+    assert_object_found_in_env(solution_code_all, env, "fail_if_code_feedback", throw_grade = FALSE)
     if (
-      is_placeholder(solution_code) ||
-        is_missing(solution_code) ||
-        is.null(solution_code) ||
-        length(solution_code) == 0 ||
-        !nzchar(solution_code)
+      is_placeholder(solution_code_all) ||
+      is_missing(solution_code_all) ||
+      is.null(solution_code_all) ||
+      length(solution_code_all) == 0 ||
+      !nzchar(solution_code_all)
```

Comment:
This needs a little more refactoring. `solution_code_all` is probably a list with class `"gradethis_solutions"`. If it's a character, we'd want to turn it into a `gradethis_solutions` (maybe?). But the check needs to be done a little differently.

Also, since we might end up calling `solutions_prepare()` on an already prepared `solution_code_all`, we should add an escape hatch there to return the object directly if it inherits the `gradethis_solutions` class.

Reply:
[d1f0ed3](https://github.com/rstudio/gradethis/pull/289/commits/d1f0ed3b591e115c56a1f5e861e7b6b09385dd03) adds an early return to `solutions_prepare()`